iml-services/iml-snapshot/src/main.rs

```diff
@@ -2,16 +2,128 @@
 // Use of this source code is governed by a MIT-style
 // license that can be found in the LICENSE file.
 
-use futures_util::stream::TryStreamExt;
+use futures::{StreamExt, TryStreamExt};
+use iml_command_utils::{wait_for_cmds_success, CmdUtilError};
+use iml_graphql_queries::snapshot as snapshot_queries;
+use iml_manager_client::{get_influx, graphql, Client, ImlManagerClientError};
 use iml_manager_env::get_pool_limit;
-use iml_postgres::{get_db_pool, sqlx};
+use iml_postgres::{get_db_pool, sqlx, PgPool};
 use iml_service_queue::service_queue::consume_data;
 use iml_tracing::tracing;
 use iml_wire_types::snapshot;
+use std::{collections::HashMap, fmt::Debug};
+use thiserror::Error;
+use tokio::{
+    time::Instant,
+    time::{interval, Duration},
+};
 
 // Default pool limit if not overridden by POOL_LIMIT
 const DEFAULT_POOL_LIMIT: u32 = 2;
 
+#[derive(Error, Debug)]
+enum Error {
+    #[error(transparent)]
+    CmdUtilError(#[from] CmdUtilError),
+    #[error(transparent)]
+    ImlManagerClientError(#[from] ImlManagerClientError),
+    #[error(transparent)]
+    ImlGraphqlQueriesError(#[from] iml_graphql_queries::Errors),
+    #[error(transparent)]
+    ImlPostgresError(#[from] iml_postgres::sqlx::Error),
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+enum State {
+    Monitoring(u64),
+    CountingDown(Instant),
+}
+
+async fn tick(snapshot_client_counts: &mut HashMap<i32, State>, pool: PgPool) -> Result<(), Error> {
+    let client: Client = iml_manager_client::get_client()?;
+    let client_2 = client.clone();
+
+    let query = iml_influx::filesystems::query();
+    let stats_fut =
+        get_influx::<iml_influx::filesystems::InfluxResponse>(client, "iml_stats", query.as_str());
+
+    let influx_resp = stats_fut.await?;
+    let stats = iml_influx::filesystems::Response::from(influx_resp);
+
+    tracing::debug!("Influx stats: {:?}", stats);
+
+    let snapshots =
+        sqlx::query!("SELECT id, filesystem_name, snapshot_name, snapshot_fsname FROM snapshot")
+            .fetch_all(&pool)
+            .await?;
+    tracing::debug!("Fetched {} snapshots from DB", snapshots.len());
+
+    for snapshot in snapshots {
+        let snapshot_id = snapshot.id;
+        if let Some(snapshot_stats) = stats.get(&snapshot.snapshot_fsname) {
```

Comment:
nit: one level of nesting could be removed by doing this:

```suggestion
        let snapshot_stats = match stats.get(&snapshot.snapshot_fsname) {
            Some(x) => x,
            None => continue,
        };
```

Reply:
Ok, done